### DISCUSSION TITLE
build in tools

### DISCUSSION THREAD

**Discussion by eishwezin20** [tensorflow/decision-forests#192] [Q&A] [answered]
Are there any built-in tools for hyperparameter tuning in TensorFlow Decision Forests?
Can I use TensorFlow Decision Forests for ranking problems?

**Comment by rstz** [comment, marked as answer]
Yes to both:

[Here is a tutorial on hyperparameter tuning](https://www.tensorflow.org/decision_forests/tutorials/automatic_tuning_colab)

[Here is a tutorial on ranking with TF-DF](https://www.tensorflow.org/decision_forests/tutorials/ranking_colab)

_Note: Our team deleted an incorrect answer to this question_
